client/components/infinite-list/index.jsx

```diff
@@ -154,9 +159,12 @@ export default class InfiniteList extends React.Component {
 		// we may have guessed item heights wrong - now we have real heights
 		if ( ! this.isScrolling ) {
 			this.cancelAnimationFrame();
-			this.updateScroll( {
-				triggeredByScroll: false,
-			} );
+			// updateScroll misbehaves when it's called syncronously from componentDidUpdate.
+			setTimeout( () => {
+				this.updateScroll( {
```

Comment:
this could also use a `_isMounted` check, in case the component unmounted after scheduling the run.